prow/github/client.go

```diff
@@ -246,22 +243,22 @@ func (c *Client) doRequest(method, path, accept string, body interface{}) (*http
 	return c.client.Do(req)
 }
 
-func (c *Client) BotName() (string, error) {
-	c.mut.Lock()
-	defer c.mut.Unlock()
-	if c.botName == "" {
```

Comment:
It seems that we already do what I asked for :)

Reply:
Whoops~ :sweat_smile:

Reply:
This is a minor refactoring anyway (don't need lock any more). Does it make any sense? Feel free to close it if you think it unnecessary :)

Reply:
You still need the lock because different goroutines in the same process are going to access this. Sorry for not looking at how this was implemented in the first place but it's already what I wanted. Thanks anyway for taking a stab at it!

/close

Reply:
Agree on closing it. But btw I don't think the lock is needed if there is only read operation as what I modified :)

Reply:
You cannot guarantee that everyone using our github library is going to do the same thing.